spec/tbd_tests_spec.rb

```diff
@@ -5206,79 +5335,113 @@
+    ptz.each { |p| vec << t * OpenStudio::Point3d.new(p.x, p.y, p.z) }
+
+    # # This window would have 2 shared edges (@right angle) with the parent.
+    expect(vec[0].x).to be_within(0.01).of(-3.48)
+    expect(vec[0].y).to be_within(0.01).of(-6.03)
+    expect(vec[0].z).to be_within(0.01).of( 4.24)
+    expect(vec[1].x).to be_within(0.01).of(-1.67)
+    expect(vec[1].y).to be_within(0.01).of(-2.90)
+    expect(vec[1].z).to be_within(0.01).of( 0.63)
+    expect(vec[2].x).to be_within(0.01).of(-4.05)
+    expect(vec[2].y).to be_within(0.01).of(-7.02)
+    expect(vec[2].z).to be_within(0.01).of( 1.82)
+    expect(vec[3].x).to be_within(0.01).of(-4.67)
+    expect(vec[3].y).to be_within(0.01).of(-8.09)
+    expect(vec[3].z).to be_within(0.01).of( 3.05)
 
     # Adding a Frame & Divider object for w3.
     expect(w3.allowWindowPropertyFrameAndDivider).to be(true)
     expect(w3.setWindowPropertyFrameAndDivider(fd2)).to be(true)
     width = w3.windowPropertyFrameAndDivider.get.frameWidth
     expect(width).to be_within(0.001).of(0.200)
 
-    u, opening_area, opening_vertices = opening(fd2_model, "w3")
-    expect(opening_area).to be_a(Numeric)
-    expect(opening_vertices.nil?).to be(false)
-    expect(opening_area).to be_within(0.1).of(1.1)
-    expect(u).to be_within(0.01).of(2.0)
-    expect(opening_vertices.size).to eq(3)
-    opening_vertices = t * opening_vertices
-
-    # This window would have 2 shared edges (@right angle) with the parent.
-    expect(opening_vertices[0].x).to be_within(0.01).of(-4.26)
-    expect(opening_vertices[0].y).to be_within(0.01).of(-7.37)
-    expect(opening_vertices[0].z).to be_within(0.01).of(10.00)
-    expect(opening_vertices[1].x).to be_within(0.01).of(-5.00)
-    expect(opening_vertices[1].y).to be_within(0.01).of(-8.66)
-    expect(opening_vertices[1].z).to be_within(0.01).of( 8.52)
-    expect(opening_vertices[2].x).to be_within(0.01).of(-5.00)
-    expect(opening_vertices[2].y).to be_within(0.01).of(-8.66)
-    expect(opening_vertices[2].z).to be_within(0.01).of(10.00)
+    surface = openings(fd_model, dad)
+    expect(surface.nil?).to be(false)
+    expect(surface.is_a?(Hash)).to be(true)
+    expect(surface.has_key?(:windows)).to be(true)
+    expect(surface[:windows].is_a?(Hash)).to be(true)
+    expect(surface[:windows].has_key?("w3"))
+    expect(surface[:windows]["w3"].is_a?(Hash)).to be(true)
+    expect(surface[:windows]["w3"].has_key?(:gross)).to be(true)
+    expect(surface[:windows]["w3"][:gross]).to be_within(0.01).of(1.1)
+    expect(surface[:windows]["w3"].has_key?(:u)).to be(true)
+    expect(surface[:windows]["w3"][:u]).to be_within(0.01).of(2.0)
+    expect(surface[:windows]["w3"].has_key?(:points)).to be(true)
+
+    ptz = surface[:windows]["w3"][:points]
+    expect(ptz.size).to eq(3)
+
+    vec = OpenStudio::Point3dVector.new
+    ptz.each { |p| vec << t * OpenStudio::Point3d.new(p.x, p.y, p.z) }
+
+    # # This window would have 2 shared edges (@right angle) with the parent.
```

Comment:
I didn't look at these tests in depth but it is very good you have them :-)

Reply:
:]